Change video_demo slightly so it works on web.

ditman opened the pull request:
## Description

Modify video_demo slightly so it compiles and runs on web (with `flutter run -d chrome`, for example).

This changes:

* Short-circuit connectivity check, until that plugin becomes available for web.
* Prevent platform checks if kIsWeb is true.

## Related Issues

* https://github.com/flutter/flutter/issues/46669

## Tests

Existing tests seem to pass.

```
$ flutter test
Building flutter tool...
Running "flutter pub get" in flutter_gallery...                     0.5s
00:41 +127 ~17: /usr/local/google/home/dit/github/flutter/examples/flutter_gallery/test/smoke_test.dart: Flutter Gallery app smoke test                     
> VideoDemo initController "butterfly" 
> VideoDemo initController "bee" 
> VideoDemo dispose
< VideoDemo dispose
01:09 +128 ~17: /usr/local/google/home/dit/github/flutter/examples/flutter_gallery/test/smoke_test.dart: Flutter Gallery app smoke test with semantics      
> VideoDemo initController "butterfly" 
> VideoDemo initController "bee" 
> VideoDemo dispose
< VideoDemo dispose
01:10 +129 ~17: All tests passed!                                                                                                                           
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

ditman committed:
Change video_demo slightly so it works on web.

This changes:

* Short-circuit connectivity, until that plugin becomes available for web.
* Prevent platform checks if kIsWeb is true.

hterkelsen commented on examples/flutter_gallery/lib/demo/video_demo.dart:
```diff
@@ -306,6 +307,13 @@ class _ConnectivityOverlayState extends State<ConnectivityOverlay> {
   @override
   void initState() {
     super.initState();
+    if (kIsWeb) {
```
Add a TODO to remove this once the `connectivity` plugin is supported on web. We can support it with: https://api.dart.dev/stable/2.7.0/dart-html/Navigator/onLine.html

ditman replied:
Do we have a ticket for the connectivity plugin?

hterkelsen replied:
Just added this: https://github.com/flutter/flutter/issues/46735

ditman replied:
Added a TODO, linked to the issue above.

hterkelsen commented on examples/flutter_gallery/lib/demo/video_demo.dart:
```diff
@@ -324,7 +332,9 @@ class _ConnectivityOverlayState extends State<ConnectivityOverlay> {
   void dispose() {
-    connectivitySubscription.cancel();
+    if (connectivitySubscription != null) {
+      connectivitySubscription.cancel();
```
can be `connectivitySubscription?.cancel()`

ditman replied:
Ah, good point!

ditman replied:
Done.

ditman committed:
Address first round of comments.

ditman commented:
cc @rami-a (this may help when migrating the video demo to the gallery v2?)

sigurdm reviewed (approved):
LGTM

goderbauer commented on examples/flutter_gallery/pubspec.yaml:
```diff
@@ -14,7 +14,7 @@ dependencies:
   url_launcher: 5.2.7
   cupertino_icons: 0.1.2
-  video_player: 0.10.3+3
+  video_player: 0.10.5
```
You will have to run "flutter update-packages --force-upgrade" and include the changes in the PR when changing a pubspec.yaml.

ditman replied:
@goderbauer this makes sense! I did it but it felt it was rolling forward "too many" versions of other packages, so I held this. Also there's a new gallery coming, so I'm not sure if I should update and merge this, or not :)

goderbauer replied:
Yeah, if you update one dependency you'll have to roll all of them forward.

It's upto you. You can update this PR or close it.

ditman replied:
@goderbauer ok, I'll get this landed ASAP so I can resolve a couple of tickets.

ditman replied:
PTAL?

ditman committed:
Add .gitignore to exclude web artifacts.

ditman committed:
flutter update-packages --force-upgrade

(After bumping video_player to 0.10.5+1 in
examples/flutter_gallery/pubspec.yaml)

ditman committed:
Fix analyzer error

ditman commented:
Waiting for @goderbauer to take another look, and for the tree to become green again!

goderbauer reviewed (approved):
LGTM

ditman commented:
Yay! Thanks @fluttergithubbot! :D

flar commented:
flutter update-packages breaks with this commit.

flar commented:
Only my update-packages seem to be broken with this commit, others aren't having an issue.
